ios/Breez Notification Service Extension/NotificationService.swift

```diff
@@ -15,70 +19,54 @@ class NotificationService: UNNotificationServiceExtension {
         return log
     }()
     
-    private static var breezSDK: BlockingBreezServices?
-    private var paymentReceivers: [PaymentReceiver] = []
-    private var paymentHashPollerTimer: Timer?
+    private var breezSDK: BlockingBreezServices? = nil
     
+    private var contentHandler: ((UNNotificationContent) -> Void)? = nil
+    private var bestAttemptContent: UNMutableNotificationContent? = nil
+    private var currentTask: SDKBackgroundTask? = nil
     
     override func didReceive(
         _ request: UNNotificationRequest,
         withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void
     ) {
-        logger.info("Notification received")
-        guard let bestAttemptContent = request.content.mutableCopy() as? UNMutableNotificationContent else {
-            return
-        }
-        guard let paymentHash = bestAttemptContent.userInfo["payment_hash"] as? String else {
-            contentHandler(bestAttemptContent)
-            return
-        }
-        
-        let paymentReciever = PaymentReceiver(logger: self.logger, contentHandler: contentHandler, bestAttemptContent: bestAttemptContent, paymentHash: paymentHash)
+        Self.logger.info("Notification received")
+        self.contentHandler = contentHandler
+        self.bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
         
-        DispatchQueue.main.async {
-            self.paymentReceivers.append(paymentReciever)
-            if NotificationService.breezSDK == nil {
+        if let currentTask = self.getTaskFromNotification() {
+            self.currentTask = currentTask
+            
+            DispatchQueue.main.async {
                 do {
-                    self.logger.info("Breez SDK is not connected, connecting....")
-                    try setLogStream(logStream: SDKLogListener(logger: self.logger))
-                    NotificationService.breezSDK = try connectSDK(paymentListener: {[weak self](payment: Payment) in
-                        DispatchQueue.main.async {
-                            self?.onPaymentReceived(payment: payment)
-                        }
-                    })
-                    self.logger.info("Breez SDK connected successfully")
+                    Self.logger.info("Breez SDK is not connected, connecting....")
+                    self.breezSDK = try BreezManager.register(listener: currentTask)
+                    Self.logger.info("Breez SDK connected successfully")
                 } catch {
-                    self.logger.info("Breez SDK connections failed \(error)")
+                    Self.logger.info("Breez SDK connections failed \(error)")
```

Comment:
I'm trying to match Android service logs with iOS extension and noticed a typo & a wrong log level here

```suggestion
                    Self.logger.error("Breez SDK connection failed \(error)")
```

Reply:
Done